wqflask/wqflask/partial_correlations_views.py

```diff
@@ -0,0 +1,266 @@
+        target_traits: Tuple[dict, ...],
+        traits_list: Tuple[dict, ...], corr_method: str) -> Tuple[
+            str, dict, Tuple[dict, ...], Tuple[dict, ...], Tuple[dict, ...],
+            str]:
+    if len(target_traits) == 0:
+        flash(
+            "You must select at least one target trait.", "alert-danger")
+        return (
+            "select-targets", primary_trait, control_traits, target_traits,
+            traits_list, corr_method)
+
+    hmacs =(primary_trait["data_hmac"],) + tuple(
+        trait["data_hmac"] for trait in (control_traits + target_traits))
+    return (
+        "select-corr-method", primary_trait, control_traits, target_traits,
+        tuple(
+            trait for trait in traits_list if trait["data_hmac"] not in hmacs),
+        corr_method)
+
+def process_step_select_corr_method(
+        primary_trait: dict, control_traits: Tuple[dict, ...],
+        target_traits: Tuple[dict, ...],
+        traits_list: Tuple[dict, ...], corr_method: str) -> Tuple[
+            str, dict, Tuple[dict, ...], Tuple[dict, ...], Tuple[dict, ...],
+            str]:
+    methods = (
+        "genetic correlation, pearson's r",
+        "genetic correlation, spearman's rho",
+        "sgo literature correlation",
+        "tissue correlation, pearson's r",
+        "tissue correlation, spearman's rho")
+    if corr_method.lower() not in methods:
+        flash(
+            "Selected method is unknown.", "alert-danger")
+        return (
+            "select-corr-method", primary_trait, control_traits, target_traits,
+            traits_list, corr_method)
+
+    hmacs =(primary_trait["data_hmac"],) + tuple(
+        trait["data_hmac"] for trait in (control_traits + target_traits))
+    return (
+        "run-correlation", primary_trait, control_traits, target_traits,
+        tuple(
+            trait for trait in traits_list if trait["data_hmac"] not in hmacs),
+        corr_method)
+
+def process_step(
+        step: str, primary_trait: dict, control_traits: Tuple[dict, ...],
+        target_traits: Tuple[dict, ...], traits_list: Tuple[dict, ...],
+        corr_method: str) -> Tuple[
+            str, dict, Tuple[dict, ...], Tuple[dict, ...], Tuple[dict, ...],
+            str]:
+    processor_functions = {
+        # "select-traits": lambda arg: arg,
+        "select-primary": process_step_select_primary,
+        "select-controls": process_step_select_controls,
+        "select-targets": process_step_select_targets,
+        "select-corr-method": process_step_select_corr_method
+    }
+    return processor_functions[(step or "select-primary")](
+        primary_trait, control_traits, target_traits, traits_list, corr_method)
+
+def sequence_of_traits(trait_strs) -> Tuple[dict, ...]:
+    return tuple(filter(
+        lambda trt: trt is not None,
+        (parse_trait(tstr.strip()) for tstr in trait_strs)))
+
+def publish_target_dabases(conn, group, threshold):
```

Comment:
*nitpick:* perhaps a typo in your function name?
```suggestion
def publish_target_databases(conn, group, threshold):
```